hpy/devel/src/runtime/argparse.c

```diff
@@ -105,15 +144,100 @@ static int
 parse_item(HPyContext ctx, HPyTracker *ht, HPy current_arg, int current_arg_tmp, const char **fmt, va_list *vl, const char *err_fmt)
 {
     switch (*(*fmt)++) {
-    case 'i': {
+
+    case 'b': { /* unsigned byte -- very short int */
+        char *output = va_arg(*vl, char *);
+        _BREAK_IF_OPTIONAL(current_arg);
+        long value = HPyLong_AsLong(ctx, current_arg);
+        if (value == -1 && HPyErr_Occurred(ctx))
+            return 0;
+        if (value < 0) {
+            set_error(ctx, ctx->h_OverflowError, err_fmt,
+                "unsigned byte integer is less than minimum");
+            return 0;
+        }
+        if (value > UCHAR_MAX) {
+            set_error(ctx, ctx->h_OverflowError, err_fmt,
+                "unsigned byte integer is greater than maximum");
+            return 0;
+        }
+        *output = (char) value;
+        break;
+    }
+
+    case 'B': { /* byte sized bitfield - both signed and unsigned
+                   values allowed */
+        char *output = va_arg(*vl, char *);
+        _BREAK_IF_OPTIONAL(current_arg);
+        unsigned long value = HPyLong_AsUnsignedLongMask(ctx, current_arg);
+        if (value == (unsigned long)-1 && HPyErr_Occurred(ctx))
+            return 0;
+        *output = (unsigned char) value;
+        break;
+    }
+
+    case 'h': { /* signed short int */
+        short *output = va_arg(*vl, short *);
+        _BREAK_IF_OPTIONAL(current_arg);
+        long value = HPyLong_AsLong(ctx, current_arg);
+        if (value == -1 && HPyErr_Occurred(ctx))
```

Comment:
Missing braces?

Reply:
If you're referring to the missing `{}` then HPy already uses the pattern without the braces quite a lot for returning errors. I'm happy to change it, but I think it would be good to agree on a standard first.